server/src/main/java/org/opensearch/indices/recovery/RecoverySettings.java

```diff
@@ -159,6 +160,7 @@ public class RecoverySettings {
 
     private volatile ByteSizeValue chunkSize = DEFAULT_CHUNK_SIZE;
 
+    @Inject
```

Comment:
I don't think we need these Injects  for SegmentReplicationTargetService, we can bind that class similar to `PeerRecoverySourceService` in Node.java.  We can create a SegmentReplicationSourceFactory there too and avoid injecting it.

```
b.bind(SegmentReplicationTargetService.class)
  .toInstance(new SegmentReplicationTargetService(transportService, indicesService, recoverySettings, new SegmentReplicationSourceFactory(...)));
```